modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbTableStorage.java

```diff
@@ -254,54 +246,21 @@ public void destroy() throws StorageException {
     /** {@inheritDoc} */
     @Override
     public PartitionStorage getOrCreatePartition(int partId) throws StorageException {
-        assert !stopped : "Storage has been stopped";
-
-        synchronized (partitionsLock) {
-            checkPartitionId(partId);
-
-            if (partitions[partId] != null) {
-                return partitions[partId];
-            }
-
-            var newPartition = new RocksDbPartitionStorage(threadPool, partId, db, partitionCf);
-
-            partitions[partId] = newPartition;
-
-            try {
-                meta.putPartitionsList(getPartitionIds());
-            } catch (RocksDBException e) {
-                throw new StorageException("Unable to save a list of partition IDs", e);
-            }
-
-            return newPartition;
-        }
-    }
+        PartitionStorage storage = getPartition(partId);
 
-    /**
-     * Returns a list of existing partition IDs.
-     */
-    private int[] getPartitionIds() {
-        int nonNullPartitionsCount = 0;
-
-        for (PartitionStorage s : partitions) {
-            if (s != null) {
-                nonNullPartitionsCount += 1;
-            }
+        if (storage != null) {
+            return storage;
         }
 
-        int[] partitionIds = new int[nonNullPartitionsCount];
+        // possible races when creating the partitions with the same ID are safe, since both the storage creation and the meta update
```

Comment:
```suggestion
        // Possible races when creating the partitions with the same ID are safe, since both the storage creation and the meta update
```